### DISCUSSION TITLE
How to request RootView (conductor) to activate another view (Screen)?

### DISCUSSION THREAD

**Discussion by mwdpb** [canton7/Stylet#344] [Q&A]
Suppose we have a RootView which is a `Conductor<IScreen>.Collection.OneActive` that is managing two child views, Screen A and Screen B.  Within Screen A, there's a button callback which upon firing will switch to Screen B.  

One way is to pass RootVM to each child VMs but that's not very elegant.  I would expect a similar method call like `RequestClose`.  What's the best way to implement that?

**Comment by canton7** [comment]
You can use the event aggregator, or a child can cast its `Parent` property to some suitable interface and call that, or I've got a sample in the samples folder which uses a navigation controller (which is closer to what I do in my apps).

I'm a bit reluctant to build something in, given the diverse sets of requirements and options out there, but you should find you've got the building blocks you need to construct something easily.

**Reply by canton7** [reply]
https://github.com/canton7/Stylet/tree/master/Samples/Stylet.Samples.NavigationController

**Reply by mwdpb** [reply]
Thanks for the explanation.  I put the RootVM in a global store and get the job done.  :)
